src/static/js/contributors.js

```diff
@@ -0,0 +1,24 @@
+const style = document.querySelector('style');
+const buttons = document.querySelectorAll('.filter-container button');
+const contributors = document.querySelector('.contributors');
+
+for (let button of buttons) {
+  const buttonClassName = button.className;
+
+  // Dynamically generate CSS that reacts to contributors' view state and buttons' toggle state
```

Comment:
Rather than generating the styles in JS, you can do the iteration in the `styles` block of the contributors template, so it's pure CSS.